kivy/uix/behaviors/compoundselection.py

```diff
@@ -308,7 +308,7 @@ def select_with_touch(self, node, touch=None):
             return False
 
         if (node in self.selected_nodes and (not range_select)):  # selected
-            if multiselect:
+            if ((not multi) or multiselect):
                 self.deselect_node(node)
             else:
                 self.clear_selection()
```

Comment:
@saqib1707, removing line 315 also changes the current behavior for `touch_multiselect=False`, not just `multiselect=False`, and allows deselecting the last item by touch when touch_multiselect=False.

Though you are right about the edge case you have described, it looks like line 315 should not be deleted, but put behind an `if` block and only called if the conditions are met for your use case.